97_extra_materials.Rmd

```diff
@@ -290,3 +290,324 @@ e <- attr(reducedDim(tse_Genus, "PCoA_BC"), "eig");
+PHY_ord <- ordinate(PHY_relab, method = "PCoA", distance = "bray")
+
+library(scater)
+
+# For the ordination using mia we use runMDDS from the scater package and implement the vegan bray distance calculation
+TSE <- runMDS(TSE,
+              FUN = vegan::vegdist,
+              method = "bray",
+              assay.type = "relabundance",
+              name = "MDS_bray",
+              ncomponents = 10) # Let's also define how many dimensions
+TSE # Inspect, now we have new reducedDim "MDS_bray"
+
+```
+
+### Plotting ordinations: plot_ordination = plotReducedDim
+Phyloseq has it's own plotting fuction for ordinations.
+
+```{r}
+plot_ordination(physeq = PHY, ordination = PHY_ord, color = "SampleType")
+```
+
+It is also easy to plot the ordination stored in reducedDim in the TSE using the plotReducedDim. We can first check what the name of the Bray-Curtis MDS/PCoA was.
+```{r}
+# Check reduced dim names
+reducedDimNames(TSE)
+
+```
+Ok, let's plot.
+```{r}
+# Plot
+plotReducedDim(TSE, "MDS_bray", color_by = "SampleType")
+
+# The sign is given arbitrarily so we can change it to match the plot_ordination
+
+reducedDim(TSE)[,1] <- -reducedDim(TSE)[,1]
+reducedDim(TSE)[,2] <- -reducedDim(TSE)[,2]
+
+plotReducedDim(TSE, "MDS_bray", color_by = "SampleType")
+```
+
+
+### Agglomerating taxa: tax_glom=mergeFeaturesByRank
+
+Often you might want to study your data using different taxonomic ranks, for example check if you see differences in the abundances of higher taxonomic levels.
+
+This data object can again be conveniently stored in a TSE object under altExp.
+
+
+
+```{r}
+PHY_fam <- tax_glom(PHY, taxrank = "Family")
+
+
+```
+
+Tax glom removes the taxa which have not been assigned to the level given in taxrank by default. NArm = TRUE so we will add the onRankOnly = TRUE which is equivalent to the mergeFeaturesByRank.
+
+```{r}
+altExp(TSE, "Family") <-
+  mergeFeaturesByRank(TSE,
+                      rank = "Family",
+                      onRankOnly = TRUE, # Let's add the equivalent option to NArm =TRUE
+                      agglomerateTree = TRUE)
+altExp(TSE, "Family")
+```
+
+
+### Cheatsheet
+
+```{r}
+library(knitr)
+df <- data.frame(Functionality=c("Access sample data", "Access tax table", "Access OTU table", "Build data object", "Calculate alpha diversity", "Calculate beta diversity", "Plot ordination", "Subset taxa", "Subset samples", "Aggromerate taxa"), phyloseq=c("sample_data()", "tax_table()", "otu_table()", "phyloseq()", "estimate_richness()", "ordinate()", "plot_ordination()", "subset_taxa()", "subset_samples()", "tax_glom()"), miaTreeSE = c("Index columns", "Index rows", "assays()", "TreeSummarizedExperiment()", "estimateDiversity()", "runMDS()", "plotReducedDim()", "Index rows", "Index columns", "mergeFeaturesByRank()"))
+
+df2 <- data.frame(Data_type=c("OTU table", "Taxonomy table", "Sample data table"),phyloseq=c("otu_table", "tax_table", "sample_data"), TreeSE=c("assay","rowData","colData"))
+kable(df)
```

Comment:
Would be helpful for later edits if this can written in a way that the methods form pairs, for instance one pair per row. So one can directly see from this code easily what is matching what?